navigation/navigation-compose/build.gradle

```diff
@@ -44,12 +44,31 @@ kotlin {
 
     sourceSets {
         commonMain {
+            def isItForPublication = project.findProperty("jetbrains.publication.libraries")
+                    ?.toString()
+                    ?.contains("NAVIGATION") == true
+
             dependencies {
                 api project(":core:core-bundle")
                 implementation "org.jetbrains.compose.foundation:foundation-layout:1.7.1"
-                implementation "org.jetbrains.compose.animation:animation-core:1.7.1"
+
+                /**
+                 * TODO: https://youtrack.jetbrains.com/issue/CMP-7390/Migrate-iosInstrumentedTest-target-to-kotlin-2.1.0
+                 * Ultimately, we should publish the navigation library with pinned dependencies.
+                 * When using 1.7.x libs built with K1, some specific IR parts in those pinned artifacts are invalid.
+                 * Starting with Kotlin 2.1.0, the invalid parts are not ignored anymore.
+                 * So we will need to rebuild the dependencies and to re-pin them.
+                 * For now, to make the :mpp:demo run, let's use project dependencies.
+                 */
+                if (isItForPublication) {
+                    implementation "org.jetbrains.compose.animation:animation-core:1.7.1"
+                    api "org.jetbrains.compose.runtime:runtime:1.7.1"
```

Comment:
Isn't the issue exist with published libraries too?

Can we just use the project version?
```
implementation(project(":compose:animation:animation-core"))
api(project(":compose:runtime:runtime"))
```

The pinning guidelines are to replace it by stable version (1.8.0) when we build it later.

Reply:
The published navigation won't cause a problem because it's already an existing klib with different IR (produced by older Kotlin and compiler plugin).

> Can we just use the project version?
I think we can imo, if it's okay.

Reply:
Okay for me

Reply:
fixed